docs/user-guide/runtime/configuration.md

```diff
@@ -16,103 +17,219 @@ When running as an [embedded application](embedding), the configuration options
 
-disableEditor
-: if set to `true`, prevents the editor UI from being served by the runtime. The admin api endpoints remain active. Default: `false`.
+```javascript
+    httpAdminAuth: {user:"nol", pass:"5f4dcc3b5aa765d61d8327deb882cf99"}
+```
+
+The `pass` property is the md5 hash of the actual password. The following
+ command can be used to generate the hash:
 
-httpStatic
-: a local directory from which to serve static web content from. This content is
-  served from the top level url, `/`. When this property is used, `httpAdminRoot` must
-  also be used to make editor UI available at a path other than `/`.
+    node -e "console.log(require('crypto').createHash('md5').update('YOUR PASSWORD HERE','utf8').digest('hex'))"
 
   *Standalone only*.
 
-httpStaticAuth
-: enabled HTTP Basic Authentication on the static content. See `httpAdminAuth` for format.
+**httpRoot**
+: *Deprecated*: see `httpAdminRoot` and `httpNodeRoot`.
 
-httpNodeCors
-: enables cross-origin resource sharing for the nodes that provide HTTP endpoints,
-  as defined [here](https://github.com/troygoode/node-cors#configuration-options)
+### Runtime
 
-httpNodeMiddleware
-: an HTTP middleware function, or array of functions, that is added to all HTTP In nodes.
-  This allows whatever custom processing, such as authentication, is needed for
-  the nodes. The format of the middleware function is
-  documented [here](http://expressjs.com/guide/using-middleware.html#middleware.application).
+**lang**
+*Since Node-RED 2.0.0.*
+: the following option is to run node-red in your preferred language.<br/> 
+Available languages include: `en-US (default), ja, de, zh-CN, zh-TW, ru, ko`. Some languages are more complete than others.
 
-      httpNodeMiddleware: function(req,res,next) {
-          // Perform any processing on the request.
-          // Be sure to call next() if the request should be passed
-          // to the relevant HTTP In node.
-      }
+**diagnostics**
+*Since Node-RED 3.0.0.*
+: Configure diagnostics options.
 
-logging
+```javascript
+    diagnostics: {
+        /** @type {boolean} enable or disable diagnostics. Must be set to `false` to disable */
+        enabled: true,
+        /** @type {"basic"|"admin"} diagnostic level can be "basic" (default) or "admin" (more sensitive details are included) */
+        level: "basic",
+    },
+```
+When `enabled` is `true` (or unset), diagnostics data will be available at http://localhost:1880/diagnostics . When `level` is "basic" (or unset), the diagnostics will not include sensitive data. Set level to "admin" for detailed diagnostics.  
+
+**runtimeState**
+*Since Node-RED 3.0.0.*
+: enable or disable flows/state
+
+```javascript
+    runtimeState: {
+        /** enable or disable flows/state endpoint. Must be set to `false` to disable */
+        enabled: false,
+        /** show or hide runtime stop/start options in the node-red editor. Must be set to `false` to hide */
+        ui: false,
+    },
+```
```

Comment:
```suggestion
**runtimeState**
*Since Node-RED 3.0.0.*
: enable or disable the ability to Start and Stop the runtime

```javascript
    runtimeState: {
        /** enable or disable this feature. */
        enabled: false,
        /** show or hide runtime stop/start options in the node-red editor. */
        ui: false,
    },
```
```